src/main/java/org/opensearch/neuralsearch/processor/util/processorSearchResponseUtil.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.processor.util;
+
+import org.opensearch.common.collect.Tuple;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.search.SearchHit;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Stack;
+
+/**
+ * Utility class for evaluating SearchResponse data. This is useful when you want
+ * to see that the searchResponse is in correct form or if the data you want to extract/edit
+ * from the SearchResponse
+ */
+public class processorSearchResponseUtil {
```

Comment:
Naming convention. Also these methods are generic and can be used in any processor, not necessarily search response processor.
```suggestion
public class ProcessorUtils {
```